BatteryConditionValidation.c

```diff
@@ -0,0 +1,142 @@
+ *     \returns     Battery parameter validity status
+ *
+*//*------------------------------------------------------------------------*/
+static int checkBatteryParam_i(float minRange_f, float maxRange_f,int batParamIndex_i, int langIndex_i)
+{
+  
+  if(battCondn_s.battCondnParam_i[batParamIndex_i] < minRange_f)
+  {
+   printf("Battery parameter %s %s!\n Current Val : %d !\n", batPar[batParamIndex_i+langIndex_i], batLevel[batParamIndex_i+langIndex_i], battCondn_s.battCondnParam_i[batParamIndex_i]);
+   return 0;
+  }
+  else if(battCondn_s.battCondnParam_i[batParamIndex_i] > maxRange_f)
+  {
+   printf("Battery parameter %s %s!\n Current Val : %d !\n", batPar[batParamIndex_i+langIndex_i], batLevel[batParamIndex_i+langIndex_i], battCondn_s.battCondnParam_i[batParamIndex_i]);
+   return 0;
+  }
+  else
+  {
+    informTrendChange_v(batParamIndex_i, langIndex_i);
+    return 1;
+  }
+}
+
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION:    informTrendChange_v
+ */
+/*!    \brief       Print the change in trend when the value is Normal
+ * 
+ *     \param       batParamIndex_i[IN] - Index of the Battery parameter
+ *     \returns     void
+ *
+*//*------------------------------------------------------------------------*/
+static void informTrendChange_v(int batParamIndex_i, int langIndex_i)
+{
+   int valChange_i = 0;
+  
+   valChange_i = battCondn_s.battCondnParam_i[batParamIndex_i] - prevBattCondn_s.battCondnParam_i[batParamIndex_i];
+  
+   if(4 <= valChange_i)
+   {
+      printf(" Battery parameter %s %s.\n Current trend : Value is increasing(Approaching higher threshold) \n Current Val : %d !\n", batPar[batParamIndex_i+langIndex_i], batLevel[batParamIndex_i+langIndex_i], battCondn_s.battCondnParam_i[batParamIndex_i]);
+   }
+   else if(4 >= valChange_i)
+   {
+     printf(" Battery parameter %s %s .\n Current trend : Value is decreasing(Approaching lower threshold) \n Current Val : %d !\n", batPar[batParamIndex_i+langIndex_i], batLevel[batParamIndex_i+langIndex_i], battCondn_s.battCondnParam_i[batParamIndex_i]);
+   }
+   else
+   {
+      printf(" Battery parameter %s %s .\n Current Val : %d !\n", batPar[batParamIndex_i+langIndex_i], batLevel[batParamIndex_i+langIndex_i], battCondn_s.battCondnParam_i[batParamIndex_i]);
+   }  
+}
+
+int main() {
+  battCondn_s.battCondnParam_i[0] = 25;
+  battCondn_s.battCondnParam_i[1] = 70;
+  battCondn_s.battCondnParam_i[2] = 0.7;
+  langVal_i = GERMAN_LANGUAGE;
+  assert(testBatteryCond_i());
+  battCondn_s = prevBattCondn_s;
+  
+  battCondn_s.battCondnParam_i[0] = 50;
+  battCondn_s.battCondnParam_i[1] = 10;
+  battCondn_s.battCondnParam_i[2] = 0;
+  assert(!testBatteryCond_i());
+  langVal_i = ENGLISH_LANGUAGE;
+  battCondn_s = prevBattCondn_s;
+  
+  battCondn_s.battCondnParam_i[0] = 50;
+  battCondn_s.battCondnParam_i[1] = 75;
+  battCondn_s.battCondnParam_i[2] = 0.9;
+  langVal_i = ENGLISH_LANGUAGE;
+  assert(!testBatteryCond_i());
```

Comment:
Pass `battCondnParam_i` to `testBatteryCond_i`. Avoid globals. Pure functions are thread-safe.